app/src/main/java/com/jerboa/ui/components/settings/Settings.kt

```diff
@@ -0,0 +1,407 @@
+                    }
+                ) {
+                    Text(text = selectionOption)
+                }
+            }
+        }
+    }
+}
+
+@Composable
+fun BannerWithClose(url: String, onClick: () -> Unit) {
+    Box(contentAlignment = Alignment.TopEnd) {
+        PictrsBannerImage(url = url)
+        IconButton(onClick = onClick) {
+            Icon(imageVector = Icons.Default.Close, contentDescription = "Remove Current Banner")
+        }
+    }
+}
+
+@Composable
+fun AvatarWithClose(url: String, onClick: () -> Unit) {
+    Box(contentAlignment = Alignment.TopEnd) {
+        LargerCircularIcon(icon = url)
+        IconButton(onClick = onClick) {
+            Icon(imageVector = Icons.Default.Close, contentDescription = "Remove Current Avatar")
+        }
+    }
+}
+
+@Composable
+fun SettingsForm(
+    settingsViewModel: SettingsViewModel,
+    siteViewModel: SiteViewModel,
+    account: Account?,
+    onClickSettings: (form: SaveUserSettings) -> Unit,
+) {
+    val luv = siteViewModel.siteRes?.my_user?.local_user_view
+    val scope = rememberCoroutineScope()
+    val ctx = LocalContext.current
+    var displayName by rememberSaveable { mutableStateOf(luv?.person?.display_name.orEmpty()) }
+    var bio by rememberSaveable { mutableStateOf(luv?.person?.bio.orEmpty()) }
+    var email by rememberSaveable { mutableStateOf(luv?.local_user?.email.orEmpty()) }
+    var matrixUserId by rememberSaveable { mutableStateOf(luv?.person?.matrix_user_id.orEmpty()) }
+    val theme by rememberSaveable { mutableStateOf(luv?.local_user?.theme.orEmpty()) }
+    val lang by rememberSaveable { mutableStateOf(luv?.local_user?.lang.orEmpty()) }
+    var avatar by rememberSaveable { mutableStateOf(luv?.person?.avatar.orEmpty()) }
+    var banner by rememberSaveable { mutableStateOf(luv?.person?.banner.orEmpty()) }
+    var defaultSortType by rememberSaveable { mutableStateOf(luv?.local_user?.default_sort_type) }
+    var defaultListingType by rememberSaveable { mutableStateOf(luv?.local_user?.default_listing_type) }
+    var showAvatars by rememberSaveable { mutableStateOf(luv?.local_user?.show_avatars) }
+    var showNsfw by rememberSaveable { mutableStateOf(luv?.local_user?.show_nsfw ?: false) }
+    var showScores by rememberSaveable { mutableStateOf(luv?.local_user?.show_scores) }
+    var showBotAccount by rememberSaveable { mutableStateOf(luv?.local_user?.show_bot_accounts) }
+    var botAccount by rememberSaveable { mutableStateOf(luv?.person?.bot_account) }
+    var showReadPosts by rememberSaveable { mutableStateOf(luv?.local_user?.show_read_posts) }
+    var showNewPostNotifs by rememberSaveable { mutableStateOf(luv?.local_user?.show_new_post_notifs) }
+    var sendNotificationsToEmail by rememberSaveable { mutableStateOf(luv?.local_user?.send_notifications_to_email) }
+    val form = SaveUserSettings(
+        display_name = displayName,
+        bio = bio,
+        email = email,
+        auth = account?.jwt,
```

Comment:
Apologies, I just checked the API, auth is a required field, so change that form to `val auth: String`